### PR TITLE
fix sign-compare warning

### PR DESCRIPTION
**Pull request by lelelelelez**
### PR types

Others
### PR changes

Others
### Describe

fix sign-compare in warning 2

### CODE REVIEW

**Review by zhaocaibei123 — APPROVED** [review]

File: paddle/fluid/distributed/ps/service/brpc_ps_client.cc [hunk 1]
```diff
@@ -1778,7 +1778,7 @@ void BrpcPsClient::PushDenseTaskConsume() {
               });
           ++merge_count;
         }
-        for (int i = 0; i < merge_count; ++i) {
+        for (uint32_t i = 0; i < merge_count; ++i) {
```
Comment (zhiqiu) [inline paddle/fluid/distributed/ps/service/brpc_ps_client.cc:1781]: uint32_t -> size_t is better.

File: paddle/fluid/distributed/ps/table/memory_dense_table.cc [hunk 2]
```diff
@@ -74,14 +74,14 @@ int32_t MemoryDenseTable::InitializeValue() {
...
-    for (int y = 0; y < dim; ++y) {
+    for (size_t y = 0; y < dim; ++y) {
       values_[x][y] = initializers_[varname]->GetValue();
     }
   }
 
   fixed_len_params_dim_ = 0;
   for (int x = 0; x < size; ++x) {
-    auto& dim = common.dims()[x];
+    int dim = common.dims()[x];
```
Comment (zhiqiu) [inline paddle/fluid/distributed/ps/table/memory_dense_table.cc:84]: Better keep it and you can cast it when doing comparing.

if (static_cast<int>(dim) != param_dim_)

File: paddle/fluid/distributed/ps/table/ssd_sparse_table.cc [hunk 3]
```diff
@@ -624,7 +625,7 @@ int32_t SSDSparseTable::Load(const std::string& path,
 }
 
 //加载path目录下数据[start_idx, end_idx)
-int32_t SSDSparseTable::Load(size_t start_idx, size_t end_idx,
+int32_t SSDSparseTable::Load(size_t start_idx, int end_idx,
```
Comment (zhiqiu) [inline paddle/fluid/distributed/ps/table/ssd_sparse_table.cc:628]: Better keep size_t to be consistent with start_idx.

**Review by zhiqiu — APPROVED** [review]
LGTM, some minor comments to be fixed in the next PR.

**Pull request merged**